### ISSUE TITLE
Truncate error probably with array index

### ISSUE THREAD

**Issue by roman1720** [arangodb/arangodb#2188]
When I try to truncate collection in javascript it gives error, also truncate in web admin gives "Collection error". There are two array indexes, ipList[\*].ip and ipList[\*].net both non-unique hash. When I delete them, then collection can be truncated without error. This happened after moving from 3.1RC (worked without any problems) to official latest release.

**Comment by Simran-B** [comment]
By truncating in JS, do you mean in Arangosh, Foxx or via arangojs?

**Comment by roman1720** [comment]
arangojs, same problem in web admin, so I guess it doesn't matter from where truncate is called.

**Comment by jsteemann** [comment]
@roman1720: yes, the web admin does not use arangojs, so it's probably a server-side problem.

**Comment by jsteemann** [comment]
So far I haven't been able to reproduce the issue.
This may be due to the type of documents that I used. I tried several document structures, but to no avail.
Could you provide a few example documents from the collection in question?
Additionally, what is the error message when trying to truncate the collection with the ArangoShell? It may provide a more specific error message than just "collection error".
Thanks!

**Comment by roman1720** [comment]
Here is error:


https://www.dropbox.com/s/y8a6w0q4cuu80f0/truncate_tmp.tgz

**Comment by jsteemann** [comment]
Fails with:
```
arangod> db.vIntL3.truncate()
JavaScript exception in file '/home/steemann/Arango31NoAsan/js/server/modules/@arangodb/arango-collection.js' at 132,15: ArangoError 4: internal error
!  return this.TRUNCATE();
!              ^
stacktrace: ArangoError: internal error
    at Error (native)
    at ArangoCollection.truncate (/home/steemann/Arango31NoAsan/js/server/modules/@arangodb/arango-collection.js:132:15)
    at <shell command>:1:11
```

**Comment by m0ppers** [comment]
This bug has just been fixed and 3.1.2 is on the way :tada: should be out tomorrow or so

**Comment by roman1720** [comment]
thanks, great support as always :)
